draco/asp/hard.lp

```diff
@@ -39,8 +117,35 @@ violation(count_with_field) :-
     attribute((encoding,aggregate),E,count),
     attribute((encoding,field),E,_).
 
+% @hard(text_mark_without_text_channel) Text mark requires text encoding.
+violation(text_mark_without_text_channel) :-
+    attribute((mark,type),M,text),
+    not attribute(mark_channel,M,text).
+
+% @hard(text_channel_without_text_mark) Text channel requires text mark.
+violation(text_channel_without_text_mark) :-
+    attribute(mark_channel,M,text),
+    not attribute((mark,type),M,text).
+
 % @hard(point_tick_bar_without_x_or_y) Point, tick, and bar require x or y channel.
 violation(point_tick_bar_without_x_or_y) :-
     attribute((mark,type),M,(point;tick;bar)),
     not attribute(mark_channel,M,x),
     not attribute(mark_channel,M,y).
+
+% @hard(line_area_without_x_y) Line and area require x and y channel.
+violation(line_area_without_x_y) :-
+    attribute((mark,type),M,(line;area)),
+    not attribute(mark_x_y_channels,M).
```

Comment:
Can you write this without the helper? I think you can use either two constraints or as a shortcut use `;`. I'd like to avoid helpers that are easily replaced.